File: library/src/main/java/com/getstream/sdk/chat/rest/response/QueryChannelsResponse.java
package com.getstream.sdk.chat.rest.response;

import com.getstream.sdk.chat.model.Channel;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class QueryChannelsResponse {
    @SerializedName("channels")
    @Expose
    private List<ChannelState> channels;

    public List<ChannelState> getChannelStates() {
        return channels;
    }

    public List<Channel> getChannels() {
        List<Channel> channels = new ArrayList<Channel>();
        for (ChannelState cs : getChannelStates()) {
            channels.add(cs.getChannel());
        }
        return channels;
    }

    public void setChannelStates(List<ChannelState> channels) {
        this.channels = channels;
    }
}

File: library/src/main/java/io/getstream/chat/android/client/api/QueryChannelsResponse.kt
package io.getstream.chat.android.client.api

import com.google.gson.annotations.SerializedName
import io.getstream.chat.android.client.models.Channel
import io.getstream.chat.android.client.models.ChannelState


data class QueryChannelsResponse(
    @SerializedName("channels")
    var channelStates: List<ChannelState> = emptyList()
) {

    fun getChannels(): List<Channel> {
        return channelStates.map { state -> state.channel }
    }

}